src/ripple/shamap/SHAMapTreeNode.h

```diff
@@ -176,14 +176,43 @@ class SHAMapAbstractNode
     invariants(bool is_root = false) const = 0;
 
     static std::shared_ptr<SHAMapAbstractNode>
-    make(
-        Slice const& rawNode,
+    makeFromPrefix(
+        Slice rawNode,
+        std::uint32_t seq,
+        SHAMapHash const& hash,
+        bool hashValid,
+        beast::Journal j);
+
+    static std::shared_ptr<SHAMapAbstractNode>
+    makeFromWire(
+        Slice rawNode,
         std::uint32_t seq,
-        SHANodeFormat format,
         SHAMapHash const& hash,
         bool hashValid,
```

Comment:
It looks like all the current callers of this always pass in an invalid hash, could remove the `hash` and `hashValid` parameters.

There are several functions that use this `hash`, `hashValid` pair of params. It appears this is done to avoid a copy. I'm fine with that, but it would be good to document the rational for this on one of the functions or someone may refactor this to use `optional`.

Reply:
We could also consider using a raw pointer to the SHAMapHash as a stand in for an optional that doesn't copy.

Reply:
Hmm, interesting. Let me double-check that.

Reply:
As @seelabs  mentioned, current callers of `makeFromWire` pass in an invalid hash, both params can be removed.

Conversely, current callers of `makeFromPrefix` pass in a valid hash and `hashValid` can be removed.

I don't agree with the raw pointer idea, `SHAMapHash const&` is the convention.

Reply:
@miguelportilla Side note: The idea was to use a raw pointer to replace an optional with an expensive copy. We can't use `SHAMapHash const&` to do that (well, not without another bool parameter). But it's moot anyway, the params can be removed.